apps/OpenSpace/ext/launcher/src/sgctedit/monitorbox.cpp

```diff
@@ -0,0 +1,216 @@
+/*****************************************************************************************
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2022                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include "sgctedit/monitorbox.h"
+
+MonitorBox::MonitorBox(QRect widgetDims, std::vector<QRect> monitorResolution,
+                         unsigned int nWindows, QString* winColors)
+    : _monitorWidgetSize(widgetDims)
+    , _monitorResolution(monitorResolution)
+    , _nWindows(nWindows)
+    , _colorsForWindows(winColors)
+{
+    _nMonitors = monitorResolution.size();
+    _showLabel = (_nMonitors > 1);
+    mapMonitorResolutionToWidgetCoordinates();
+}
+
+MonitorBox::~MonitorBox()
+{
+}
+
+void MonitorBox::paintEvent(QPaintEvent *event)
+{
+    Q_UNUSED(event);
```

Comment:
Don't use this, instead just don't give the function parameter a name to suppress the warning